### PR TITLE
Feat create liked watchlater history pages

### LINKED ISSUE #11: Create liked, watchlater & history pages

### PR DESCRIPTION
**Pull request by kjais1720**
PR for Issue #11 ( closes #11 )
### Created
- Horizontal Video card
- Liked videos page
- History Page
- Watch later page
### Fixed 
- useAxios instance - The auth token was not being set in the header
- Auth-middleware - On page refresh it was redirecting to login even if user was already logged in
## Preview 
- Preview link - 
- Preview image -
- 
image

image
 
image

### CODE REVIEW

**Review by rahulcs754 — APPROVED** [review]
The code looks good and clean.
I liked you define function for watchlater and history 
Design is good

File: src/authMiddleware.js [hunk 1]
```diff
@@ -8,14 +8,12 @@ export function AuthMiddleware() {
...
+    if (!localStorage.getItem('userToken')) {
       setShowAuthModal(true)
-      navigate(-1)
-      toast.error("You need to login first!");
+      toast.error("You need to login to access this page!");
     }
-  }, []);
-  return isLoggedIn ? <Outlet /> : "";
+  },[isLoggedIn]);
+  return localStorage.getItem('userToken') ? <Outlet /> : <Navigate to="/explore" />;
```
Comment (coldpigli) [inline src/authMiddleware.js]: Liked the use of Outlet here

File: src/components/horizontalVideoCard.jsx [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+      : userResourcesDispatch({ type: ADD_TO_WATCHLATER, payload: video });
+
+  const openPlaylistModal = () => {
+    setPlaylistModalState({
+      show: true,
+      selectedVideo: video,
+    });
+  };
+
+  const dropdownButtons = [
```
Comment (coldpigli) [inline src/components/horizontalVideoCard.jsx:67]: Like how you are mapping through dropdowns, instead of hardcoding it

**Review by coldpigli — APPROVED** [review]
Liked a few things in your code that I have mentioned. 
Would love to discuss with you about axios.interceptors.request.use
Overall really clean and modular code.

**Pull request merged**